src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/MalformedRoundaboutCheck.java

```diff
@@ -150,13 +167,44 @@ public boolean validCheckForObject(final AtlasObject object)
         roundaboutEdgeSet
                 .forEach(roundaboutEdge -> this.markAsFlagged(roundaboutEdge.getIdentifier()));
 
-        // If there are car navigable edges inside the roundabout flag it, as it is probably
-        // malformed or a throughabout
-        if (!roundaboutEdgeSet.isEmpty() && roundaboutEdges != null
-                && roundaboutEdgeSet.stream().noneMatch(this::ignoreBridgeTunnelCrossings)
-                && this.roundaboutEnclosesRoads(roundaboutEdges))
+        if (!roundaboutEdgeSet.isEmpty() && roundaboutEdges != null)
         {
-            instructions.add(this.getLocalizedInstruction(0));
+            // If there are car navigable edges inside the roundabout flag it, as it is probably
+            // malformed or a throughabout
+            if (roundaboutEdgeSet.stream().noneMatch(this::ignoreBridgeTunnelCrossings)
+                    && this.roundaboutEnclosesRoads(roundaboutEdges))
+            {
+                instructions.add(this.getLocalizedInstruction(0));
+            }
+
+            final Optional<PolyLine> originalGeometryOptional = this
+                    .buildOriginalOsmWayGeometry((Edge) object);
+            if (originalGeometryOptional.isPresent())
+            {
+                final PolyLine originalGeometry = originalGeometryOptional.get();
+                // There should be a minimum amount of OSM nodes in a roundabout to have good
+                // visuals.
+                // Only count nodes when we have the full roundabout, some are split into multiple
+                // ways.
+                if (originalGeometry.size() < this.minNodes && ((Edge) object).isClosed())
```

Comment:
isClosed() only applies on particular edge, which is one section of entire roundabout. Please use isClosedWay from  utility.CommonMethods class. this method will be available after merging this [PR](https://github.com/osmlab/atlas-checks/pull/468/).